homeassistant/components/hko/weather.py

```diff
@@ -0,0 +1,76 @@
+"""Support for the HKO service."""
+from homeassistant.components.weather import (
+    Forecast,
+    WeatherEntity,
+    WeatherEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import (
+    API_CONDITION,
+    API_CURRENT,
+    API_FORECAST,
+    API_HUMIDITY,
+    API_TEMPERATURE,
+    ATTRIBUTION,
+    DOMAIN,
+    MANUFACTURER,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a HKO weather entity from a config_entry."""
+    unique_id = config_entry.unique_id
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([HKOEntity(unique_id, coordinator)], False)
+
+
+class HKOEntity(CoordinatorEntity, WeatherEntity):
+    """Define a HKO entity."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_native_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_supported_features = WeatherEntityFeature.FORECAST_DAILY
+    _attr_attribution = ATTRIBUTION
+
+    def __init__(self, unique_id_str, coordinator: DataUpdateCoordinator) -> None:
```

Comment:
```suggestion
    def __init__(self, unique_id: str, coordinator: DataUpdateCoordinator) -> None:
```
I think `config_entry.unique_id` is `str | None` so maybe you need to assert it first

Reply:
Thanks for your review! I am not sure about this one, could you give an example of asserting this correctly?

Reply:
in the `async_setup_entry` you could have an `assert config_entry.unique_id` before passing it to the entity